### PR TITLE
Update coffee extension

### LINKED ISSUE #9239: [Coffee] One click menubar toggle
### Extension

https://www.raycast.com/mooxl/coffee

### Description

Would be great if the menubar could be toggled in 1 click (like the og Caffeine app)

### Who will benefit from this feature?

Users

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by xilopaint**
## Description

Fixed #9239 #11463 #11559 #11625 #12120

- Replaced the form in the `Caffeinate For` command with arguments in the root search.
- Replaced the emojis in the `Caffeinate Status` command with Unicode characters for enhanced design consistency.
- Updated some menu bar icons.
- Fixed an issue where it was not possible to decaffeinate the Mac via the `Caffeinate Status Menu Bar`.
- Fixed an issue where the menu bar icon was not immediately updated when the Mac was caffeinated.
- Fixed the `Caffeinate While` command not working.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/coffee/src/utils.ts [hunk 1]
```diff
@@ -32,22 +32,22 @@ async function update(updates: Updates, caffeinated: boolean) {
...
-
+  exec(`/usr/bin/caffeinate ${generateArgs(additionalArgs)} || true`);
+  await update(updates, true);
   if (hudMessage) {
     await showHUD(hudMessage);
   }
+}
 
-  execSync(`/usr/bin/caffeinate ${generateArgs(additionalArgs)} || true`);
+export function isValidInteger(value: string | undefined): boolean {
```
Comment (aliaksandrZh) [inline extensions/coffee/src/utils.ts:51]: Can't we use a ```Number.isInteger()```?

Comment (xilopaint) [reply extensions/coffee/src/utils.ts]: Done.

**Comment by aliaksandrZh** [comment]
Nice work ! LGTM

**Review by pernielsentikaer — COMMENTED** [review]
The only think I can come up with is that users who have selected an icon for the menubar which is now deleted might end up with an empty icon

Can we either fallback to something, or map to old names so they also "works"

**Comment by xilopaint** [comment]
> The only think I can come up with is that users who have selected an icon for the menubar which is now deleted might end up with an empty icon

Are you sure? In my test the default icon is set when the value of the selected icon is changed in `package.json`.

**Comment by xilopaint** [comment]
> The only think I can come up with is that users who have selected an icon for the menubar which is now deleted might end up with an empty icon

@pernielsentikaer Are you sure? In my test the default icon is set when the value of the selected icon is changed in `package.json`.

**Comment by pernielsentikaer** [comment]
Should we keep a mapped array for the old names so users don't suddenly see a new icon? Could that work in any way?

**Comment by xilopaint** [comment]
> Should we keep a mapped array for the old names so users don't suddenly see a new icon? Could that work in any way?

I don't think it's a necessity since the user won't be left without an icon. Additionally, users who are no longer using the default icon have already gone through the experience of having to change it once and I believe it won't be a big trouble to do it again once they notice their icon has changed. That said, I believe the small benefit of avoiding the icon change to the default does not outweigh the inconvenience of adding complexity to the code that may not be immediately clear to contributors.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by edheltzel** [comment]
@xilopaint Love the extension, but the user experience is suffering with this update:
> "Replaced the form in the `Caffeinate For` command with arguments in the root search."

This is just a "me" use case, but I configured a keyboard shortcut (hyperkey+7) to the `Caffeinate For`. This was possible only because of the form input. Now with the arguments having to be passed through the root search, its not possible. 

Error - Screenshot 2024-05-28 at 8 01 00 AM

This Update - Screenshot 2024-05-28 at 8 04 30 AM

**Comment by pernielsentikaer** [comment]
I also think the new approach is better, but I can see the issue since the parameters in the command are optional. One thing we could do is add some preferences that set the default caffeinate time, so it will use the default values if nothing is specified.

We could also add another command (which is disabled by default) to bring back the form view. Could that work? 🙂

**Comment by edheltzel** [comment]
both are great solutions. I personally would enjoy the later of the two having the option to bring back the form. I definitely appreciate it.

**Comment by xilopaint** [comment]
I don't like the idea of bringing back the form. The problem here is that all the arguments are individually optional but filling in at least one of them is required.

@pernielsentikaer, would it be possible to work with the team on this type of validation (at least one argument must be filled) so that the hotkey brings the focus to the first argument instead of trying to run the command?

**Comment by pernielsentikaer** [comment]
I can't promise you anything, but yes, I'll try to bring it up with the team if we can figure out a way to achieve this without changing too much. This is also a niche case but I agree with @edheltzel that it doesn't work very well at the moment.

I'll get back to you 🙂

**Comment by pernielsentikaer** [comment]
@xilopaint, it will stay as is at the moment. We added it to the backlog and will see if it pops up more regularly 🙂

We could do one of the following:

-  Idea 1) If no fields are filled, we show the form
-  Idea 2) Just have one argument where you can write something like `1h 30m` or `1:30` instead

Do you have any input here, @edheltzel

**Comment by edheltzel** [comment]
Idea 1 is my preferred... And seems to be the least amount of work to achieve. 

Other than that I do appreciate everything.

**Pull request merged**